### ISSUE TITLE
Unable to resolve the id field of a harness environment data source 

### ISSUE THREAD

**Issue by abrahamkoshy** [harness-io/terraform-provider-harness#79]
When running a terraform plan that creates a harness_infrastructure_definition resource using the harness_environment data source for an environment that is not yet created, the plan fails with "The argument env_id is required but no definition was found." 

The same fix should be applied for other data sources like the harness_application.

### Terraform Version
v0.14.11

### Affected Data Source(s)
- harness_environment
- harness_application

### Debug Output
The argument env_id is required but no definition was found

### Expected Behavior
The plan should have succeeded with the env_id field set to be computed during the apply run.

### Actual Behavior
The plan failed because no definition was found for the env_id field.

### Steps to Reproduce
1. `terraform plan`

### References
https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ec2/host_data_source.go#L37-L40

https://github.com/harness-io/terraform-provider-harness/blob/main/internal/service/cd/environment/environment_data_source.go#L20

**Comment by micahlmartin** [comment]
@abrahamkoshy circling back on this. I took the example you sent me. I couldn't quite figure out where you were actually creating the environments, but could see them being referecend in the data resrouce. So it looked like you were using the `resource "harness_environment" "environments"` in one step and then looking them up in the next step with `data "harness_environment" "infra_def_environments"`. Instead you could just use the outputs of the creation step in the `harness_infrastructure_definition` resource. Here's the working example:

```
terraform {
  required_providers {
    harness = {
      source  = "harness-io/harness"
      version = "~> 0.1.13"
    }
  }
}

provider "harness" {
  # Configuration options
  # For this demo the configuration is set through environment variables
}

resource "harness_user_group" "test" {
  name = "micahtesting"

  lifecycle {
    ignore_changes = [permissions]
  }
}

variable "environments" {
  type = list(object({
    name = string
    type = string
  }))

  description = "The list of environments to create in the Harness application"

  default = [{
    name = "dev"
    type = "NON_PROD"
  }]
}

variable "infrastructure_definitions" {
  type = list(object({
    environment_name    = string
    cloud_provider_name = string
    namespace_name      = string
    release_name        = string
  }))

  description = "The list of infrastructure definitions to create in the Harness application"

  default = [{
    environment_name    = "dev"
    cloud_provider_name = "auto-kubernetes"
    namespace_name      = "default"
    release_name        = "default"
  }]
}

resource "harness_application" "harness_app" {
  name = "micahtest"
}

resource "harness_environment" "environments" {
  count  = length(var.infrastructure_definitions)
  name   = var.environments[count.index].name
  type   = var.environments[count.index].type
  app_id = harness_application.harness_app.id
}

// data "harness_environment" "infra_def_environments" {
//   count  = length(var.infrastructure_definitions)
//   app_id = harness_application.harness_app.id
//   name   = var.infrastructure_definitions[count.index].environment_name
//   depends_on = [harness_environment.environments]
// }

resource "harness_infrastructure_definition" "k8s_infra_definitions" {
  count  = length(var.infrastructure_definitions)
  name   = var.infrastructure_definitions[count.index].namespace_name
  app_id = harness_application.harness_app.id
  // env_id              = data.harness_environment.infra_def_environments[count.index].id
  env_id             = harness_environment.environments[count.index].id
  cloud_provider_type = "KUBERNETES_CLUSTER"
  deployment_type     = "KUBERNETES"

  kubernetes {
    cloud_provider_name = var.infrastructure_definitions[count.index].cloud_provider_name
    namespace           = var.infrastructure_definitions[count.index].namespace_name
    release_name        = var.infrastructure_definitions[count.index].release_name
  }
}
```

Let me know if that makes sense.

**Comment by micahlmartin** [comment]
@abrahamkoshy this should be fixed in 0.1.14
